### PR TITLE
Add aten div op

### PR DESCRIPTION
**Pull request by bszmelcz**
## Details:
- [x] add layer test for `aten::div` operator
## Ticket:
94304

### CODE REVIEW

**Review by sikorsl1 — APPROVED** [review]
Really nice piece of code :)

File: tests/layer_tests/pytorch_tests/test_div.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        np.random.randn(5).astype(np.float32),
+        np.random.randn(5, 5, 1).astype(np.float32),
+        np.random.randn(1, 1, 5, 5).astype(np.float32),
+    ]))
+    @pytest.mark.parametrize('other_tensor', ([
+        np.array([[0.5]]).astype(np.float32),
+        np.random.randn(5).astype(np.float32),
+        np.random.randn(5, 1).astype(np.float32),
+        np.random.randn(1, 5).astype(np.float32),
+    ]))
```
Comment (slyalin) [inline tests/layer_tests/pytorch_tests/test_div.py]: float32 is tested only, it doesn't expose possible issues with our conversion when https://pytorch.org/docs/stable/tensor_attributes.html#type-promotion-doc is used. Actually it is relevant to other basic arithmetic ops according to the doc. I think we should implement tests for those case and let them fail. And implement more sophisticates support in conversion functions eventually.

Comment (bszmelcz) [reply tests/layer_tests/pytorch_tests/test_div.py]: @slyalin I have added tests with not matching types which as expected failed.

Comment (eaidova) [reply tests/layer_tests/pytorch_tests/test_div.py]: I believe initial task was extend op to support such behavior, not only tests

Comment (eaidova) [reply tests/layer_tests/pytorch_tests/test_div.py]: another thing, it is TO DO in operation which also required to be fixes for claiming op supported:

>  // TODO: support mode "trunc", need to extend openvino opset for that 

so, why you add only tests?

Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_div.py]: @bszmelcz agree with @eaidova, based on ONNX export you can use `Convert` to `int64` to support `trunc` divide

Comment (bszmelcz) [reply tests/layer_tests/pytorch_tests/test_div.py]: Ok I will take care of that

**Review by slyalin — APPROVED** [review]

**Pull request merged**